osu.Game/Users/UserGridPanel.cs

```diff
@@ -4,139 +4,232 @@
+                                        {
+                                            Size = new Vector2(50),
+                                            Anchor = Anchor.Centre,
+                                            Origin = Anchor.Centre
                                         }
                                     }
                                 }
                             }
-                        },
-                        new[]
+                        }
+                    },
+                    new GridContainer
+                    {
+                        Name = "Bottom content",
+                        Margin = new MarginPadding { Top = main_panel_heigth },
+                        Height = 30,
+                        RelativeSizeAxes = Axes.X,
+                        ColumnDimensions = new[]
                         {
-                            Empty(),
-                            Empty()
+                            new Dimension(GridSizeMode.Absolute, 60),
+                            new Dimension()
                         },
-                        new Drawable[]
+                        RowDimensions = new[] { new Dimension() },
+                        Content = new[]
                         {
-                            CreateStatusIcon().With(icon =>
-                            {
-                                icon.Anchor = Anchor.Centre;
-                                icon.Origin = Anchor.Centre;
-                            }),
-                            CreateStatusMessage(false).With(message =>
+                            new Drawable[]
                             {
-                                message.Anchor = Anchor.CentreLeft;
-                                message.Origin = Anchor.CentreLeft;
-                                message.Margin = new MarginPadding { Left = margin };
-                            })
+                                CreateStatusIcon().With(icon =>
+                                {
+                                    icon.Anchor = Anchor.Centre;
+                                    icon.Origin = Anchor.Centre;
+                                    icon.Size = new Vector2(14);
+                                }),
+                                CreateStatusMessage(false).With(message =>
+                                {
+                                    message.Anchor = Anchor.CentreLeft;
+                                    message.Origin = Anchor.CentreLeft;
+                                    message.Margin = new MarginPadding { Left = 18 };
+                                })
+                            }
                         }
                     }
                 }
             };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            var ruleset = rulesets?.GetRuleset(User.PlayMode);
+
+            level.User.Value = new UserProfileData(User, ruleset.AsNonNull());
```

Comment:
`UserProfileData` was not intended for reuse anywhere else, it's a model specific to the user profile overlay. The level badge should likely just accept the `LevelInfo`.

Also this line does `ruleset.AsNonNull()`, which is subsequently contradicted by a `ruleset != null` check.